aries_cloudagent/protocols/present_proof/dif/pres_exch_handler.py

```diff
@@ -219,14 +239,82 @@ async def filter_constraints(
         if not applicable:
             continue
 
-        # TODO: create new credential with selective disclosure
-        if constraints.limit_disclosure or predicate:
-            raise PresentationExchError("Not yet implemented - createNewCredential")
+        if constraints.limit_disclosure:
+            credential_dict = json.loads(credential.cred_value)
+            new_credential_dict = reveal_doc(
+                credential_dict=credential_dict, constraints=constraints
+            )
 
+            # Using custom_document_loader for testing
+            signed_new_credential_dict = await derive_credential(
+                credential=credential_dict,
+                reveal_document=new_credential_dict,
+                suite=suite,
+                document_loader=document_loader,
+            )
+            credential = VCRecord.deserialize_jsonld_cred(
+                json.dumps(signed_new_credential_dict)
+            )
```

Comment:
I think most deserialization methods can take either string or dict. This way we don't have to go from dict -> string -> dict -> class instance